### PR TITLE
[WJ-908] Search field placeholder, general sidebar/navbar madness

### PR DESCRIPTION
**Pull request by Monkatraz**
This PR was originally just me trying to add a search field into the navbar, but it kind of got out of control.

Changes:
- Replaced the `svg_use` function with `sprite` to match the rest of the frontend
- Made the ftml code blocks use the new `sprite` functionality
- Allowed buttons to support using `ui.svg` sprites
- Simplified `ui.svg` by making every `viewbox` `0 0 24 24` unless it's `wj-karma`
- Added a `X` sprite
- Allowed `TextInput` to be clearable, among other things
- Added `DisplayObserver` so that `wj-component-loader` could determine if it should load based on if it is visible in the DOM
- The sidebar now duplicates the navbar links and dropdowns, and only displays these when the screen is small
- Renamed the classes in `ClientStatus` to match the actual name of the component
- Added a close button to the sidebar
- Added a placeholder `Search` component into the navbar

### CODE REVIEW

File: ftml/src/render/html/builder.rs [hunk 1]
```diff
@@ -83,21 +83,24 @@ where
...
-    pub fn svg_use(self, href: &'t str, viewbox: &'t str) {
+    /// Creates an inline `<svg>` using the `ui.svg` spritesheet.
+    pub fn sprite(self, id: &'t str) {
+        let viewbox = match id {
+            "wj-karma" => "0 0 64 114",
+            _ => "0 0 24 24",
+        };
+
+        let class = format!("wj-sprite sprite-{}", id);
+
```
Comment (emmiegit) [inline ftml/src/render/html/builder.rs]:
```suggestion
```

File: web/modules/components/src/Button.svelte [hunk 2]
```diff
@@ -70,7 +71,13 @@
     class:is-sharp={sharp}
     {...$$restProps}
   >
-    {#if i}<Icon {i} size="1em" />{:else}<slot />{/if}
+    {#if i?.startsWith("wj-")}
+      <Sprite {i} />
+    {:else if i}
+      <Icon {i} size="1em" />
```
Comment (emmiegit) [inline web/modules/components/src/Button.svelte:77]: Why wouldn't icons start with `wj-`? I don't understand the if logic here.

File: web/modules/ftml-components/src/components/code/_code.scss [hunk 3]
```diff
@@ -44,21 +44,26 @@
...
-/* Makes the button have a checkmark when clicked. Icon is "fluent:checkmark-12-filled",
-   see https://icon-sets.iconify.design/fluent/checkmark-12-filled/ (MIT)
-*/
-.wj-code-copy.wj-code-copy-success {
-  background-image: url("data:image/svg+xml,%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20xmlns%3Axlink%3D%22http%3A%2F%2Fwww.w3.org%2F1999%2Fxlink%22%20width%3D%221em%22%20height%3D%221em%22%20preserveAspectRatio%3D%22xMidYMid%20meet%22%20viewBox%3D%220%200%2012%2012%22%3E%3Cg%20fill%3D%22none%22%3E%3Cpath%20d%3D%22M9.765%203.205a.75.75%200%200%201%20.03%201.06l-4.25%204.5a.75.75%200%200%201-1.075.015L2.22%206.53a.75.75%200%200%201%201.06-1.06l1.705%201.704l3.72-3.939a.75.75%200%200%201%201.06-.03z%22%20fill%3D%22%2307a%22%2F%3E%3C%2Fg%3E%3C%2Fsvg%3E");
+  > .sprite-wj-clipboard-success {
+    display: none;
+  }
+
+  &.wj-code-copy-success {
```
Comment (emmiegit) [inline web/modules/ftml-components/src/components/code/_code.scss:61]: So clean... :relieved:

**Review by emmiegit — APPROVED** [review]

File: web/modules/components/src/Button.svelte (hunk 2 above)
Comment (Monkatraz) [reply web/modules/components/src/Button.svelte:77]: Remember that we have Iconify - so we can use icons like `{namespace}:{icon}`. I'm still debating whether every icon should be in our spritesheet or if Iconify should be used. I'm leaning towards the former, but Iconify makes it very easy to prototype so I'd like to keep support for it.

**Comment by Monkatraz** [comment]
thanks @ammongit

**Pull request merged**